src/backend/marsha/core/xapi.py

```diff
@@ -69,6 +69,21 @@ def __init__(self, video, statement, lti_user):
             {"contextActivities": {"category": [{"id": "https://w3id.org/xapi/video"}]}}
         )
 
+        if jwt_token.payload.get("context_id"):
+            statement["context"]["contextActivities"].update(
+                {
+                    "parent": [
+                        {
+                            "id": jwt_token.payload["context_id"],
+                            "objectType": "Activity",
+                            "definition": {
+                                "type": "http://adlnet.gov/expapi/activities/course"
+                            },
+                        }
+                    ]
+                }
+            )
```

Comment:
Can you log the alternative situation?

```suggestion
            )
        else:
            logger.warning("Statement %s for user %s has no context ID", statement["id"], user_id)
```

Reply:
Log where ? If it's just a log in marsha I see no interest, this is not an error. If it's in LDP then it's not here that it should be logged.

Reply:
I just don't want to be blind about this. That said, it can be post-analyzed _via_ context-id-less statements... Unless @sampaccoud has something to say, let's forget about my initial comment.